Shared/Samples/Snap geometry edits/SnapGeometryEditsView.GeometryEditorMenu.swift

```diff
@@ -0,0 +1,155 @@
+// Copyright 2024 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+extension SnapGeometryEditsView {
+    /// A view that provides a menu for geometry editor functionality.
+    struct GeometryEditorMenu: View {
```

Comment:
It might be not ideal, but we sometimes have to duplicate code across multiple samples instead of reusing them. Here is an excerpt from Jenn Merritt

> My understanding is that any ArcGIS API shouldn't be reused across multiple samples so that they are individual and copy paste-able as you say. I think it's one of those situations where for the sample viewer app it's not necessarily best practice, but it keeps the samples separate if a user was really only interested in one area, rather than looking at the samples/sample viewer app as a whole. Then I think more sample viewer app/platform specific things like data management or other utils are fine to reuse. My 2c!

With that said, I do think geometry editor menu can be a good candidate for a toolkit component. Can you add a comment in `common-toolkit/issues/42` to summarize our findings in the latest implementation?

Reply:
actually, I'll do this.